### DISCUSSION TITLE
tsl error expected PEM encoded private key

### DISCUSSION THREAD

**Discussion by IBouarar** [gravitational/teleport#16403] [Q&A]
I followed the instruction to get teleport installed on ubuntu laptop.
Everything worked fine until yesterday when I started to get this error when running tsh:
ERROR: expected PEM encoded private key

Here what gives me tsh login -d:
[KEYSTORE]  ERRO expected PEM encoded private key client/keystore.go:312
[CLIENT]    DEBU expected PEM encoded private key client/api.go:1056

ERROR REPORT:
Original Error: *trace.BadParameterError expected PEM encoded private key
Stack Trace:
        /go/src/github.com/gravitational/teleport/api/utils/keys/privatekey.go:166 github.com/gravitational/teleport/api/utils/keys.ParsePrivateKey
        /go/src/github.com/gravitational/teleport/api/utils/keys/privatekey.go:218 github.com/gravitational/teleport/api/utils/keys.ParseKeyPair
        /go/src/github.com/gravitational/teleport/api/utils/keys/privatekey.go:209 github.com/gravitational/teleport/api/utils/keys.LoadKeyPair
        /go/src/github.com/gravitational/teleport/lib/client/keystore.go:310 github.com/gravitational/teleport/lib/client.(*FSLocalKeyStore).GetKey
        /go/src/github.com/gravitational/teleport/lib/client/api.go:945 github.com/gravitational/teleport/lib/client.ReadProfileStatus
        /go/src/github.com/gravitational/teleport/lib/client/api.go:1054 github.com/gravitational/teleport/lib/client.Status
        /go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:1343 main.onLogin
        /go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:994 main.Run
        /go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:444 main.main
        /opt/go/src/runtime/proc.go:250 runtime.main
        /opt/go/src/runtime/asm_amd64.s:1571 runtime.goexit
User Message: expected PEM encoded private key

Any help to get this fixed?
thanks

**Comment by webvictim** [comment]
I'd clear your `~/.tsh` directory and try logging in again with `tsh login`. It's possible that something there has been corrupted.

**Comment by IBouarar** [comment]
It worked!
Thank you so much 🥇
